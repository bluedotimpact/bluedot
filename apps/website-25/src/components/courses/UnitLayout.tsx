import React from 'react';
import { isMobile } from 'react-device-detect';
import {
  HeroSection,
  HeroH1,
  Section,
  CTALinkOrButton,
  Breadcrumbs,
} from '@bluedot/ui';
import { HeroMiniTitle } from '@bluedot/ui/src/HeroSection';
import Head from 'next/head';
import SideBar from './SideBar';
import { Unit } from '../../lib/api/db/tables';
import MarkdownExtendedRenderer from './MarkdownExtendedRenderer';
import Congratulations from './Congratulations';
import { ROUTES } from '../../lib/routes';

type UnitLayoutProps = {
  // Required
  unit: Unit;
  unitNumber: number;
  units: Unit[];
};

const UnitLayout: React.FC<UnitLayoutProps> = ({
  unit,
  unitNumber,
  units,
}) => {
  const nextUnit = units[units.findIndex((u) => u === unit) + 1];

  if (!unit) {
    // Should never happen
    throw new Error('Unit not found');
  }

  return (
    <div>
      <Head>
        <title>{`${unit.courseTitle}: Unit ${unitNumber}`}</title>
        <meta name="description" content={unit.title} />
      </Head>
      <HeroSection className="unit__hero">
        <HeroMiniTitle>{unit.courseTitle}</HeroMiniTitle>
        <HeroH1>{unit.title}</HeroH1>
      </HeroSection>
      <Breadcrumbs
        className="unit__breadcrumbs sticky top-[72px] md:top-[100px] z-10"
        route={{
          title: unit.courseTitle,
          url: unit.coursePath,
          parentPages: [ROUTES.home, ROUTES.courses],
        }}
      />
      <Section className="unit__main">
        <div className="unit__content-container flex flex-col md:flex-row gap-16">
          {!isMobile && (
            <SideBar units={units} currentUnitNumber={unitNumber} />
          )}
          <div className="unit__content flex flex-col flex-1 max-w-[728px] gap-4">
            <MarkdownExtendedRenderer>
              {unit.content}
            </MarkdownExtendedRenderer>

            {nextUnit ? (
              <CTALinkOrButton className="unit__cta-link self-end mt-6" url={nextUnit.path}>
                Next unit
              </CTALinkOrButton>
            ) : (
              <Congratulations courseTitle={unit.courseTitle} courseUrl={unit.coursePath} />
            )}
          </div>
        </div>
      </Section>
    </div>
  );
};

export default UnitLayout;
