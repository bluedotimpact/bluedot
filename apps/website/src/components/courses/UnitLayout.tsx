import React, { useState, useEffect } from 'react';
import Head from 'next/head';
import clsx from 'clsx';
import { useRouter } from 'next/router';
import {
  Section,
  CTALinkOrButton,
  Breadcrumbs,
} from '@bluedot/ui';
import { FaArrowRight, FaArrowLeft } from 'react-icons/fa6';

import SideBar from './SideBar';
import { Unit, Chunk } from '../../lib/api/db/tables';
import MarkdownExtendedRenderer from './MarkdownExtendedRenderer';
import Congratulations from './Congratulations';
import { ROUTES } from '../../lib/routes';
import UnitFeedback from './UnitFeedback';
import CertificateLinkCard from './CertificateLinkCard';
import { A, H1, P } from '../Text';

type UnitLayoutProps = {
  // Required
  chunks: Chunk[];
  unit: Unit;
  unitNumber: number;
  units: Unit[];
};

type MobileHeaderProps = {
  chunks: Chunk[];
  currentChunkIndex: number;
  className?: string;
  unit: Unit;
  prevUnit?: Unit;
  nextUnit?: Unit;
  onPrevClick: () => void;
  onNextClick: () => void;
};

const MobileHeader: React.FC<MobileHeaderProps> = ({
  className,
  unit,
  chunks,
  currentChunkIndex,
  prevUnit,
  nextUnit,
  onPrevClick,
  onNextClick,
}) => {
  return (
    <div className={clsx('mobile-unit-header bg-color-canvas border-b border-color-divider w-full p-3', className)}>
      <nav className="mobile-unit-header__nav flex flex-row justify-between">
        <A className="mobile-unit-header__prev-unit-cta flex flex-row items-center gap-1 no-underline disabled:opacity-50" disabled={!prevUnit} onClick={onPrevClick} aria-label="Previous unit">
          <img src="/icons/bubble-arrow.svg" alt="" className="size-8" />
        </A>
        <div className="mobile-unit-header__course-container flex flex-row gap-2 items-center">
          <img src="/icons/course.svg" className="size-8" alt="" />
          <div className="mobile-unit-header__course-title-container flex flex-col">
            <p className="mobile-unit-header__course-header text-size-xxs text-[#999eb3]">{unit.courseTitle}</p>
            <p className="mobile-unit-header__course-title bluedot-h4 text-size-xs">{chunks[currentChunkIndex]?.chunkTitle}</p>
          </div>
        </div>
        <A className="mobile-unit-header__next-unit-cta flex flex-row items-center gap-1 no-underline disabled:opacity-50" disabled={!nextUnit} onClick={onNextClick} aria-label="Next unit">
          <img src="/icons/bubble-arrow.svg" alt="" className="size-8 rotate-180" />
        </A>
      </nav>
    </div>
  );
};

const UnitLayout: React.FC<UnitLayoutProps> = ({
  chunks,
  unit,
  unitNumber,
  units,
}) => {
  const router = useRouter();
  const [currentChunkIndex, setCurrentChunkIndex] = useState(0);
  const unitArrIndex = units.findIndex((u) => u.id === unit.id);

  const isFirstChunk = currentChunkIndex === 0;
  const isLastChunk = currentChunkIndex === chunks.length - 1;

  const nextUnit = units[unitArrIndex + 1];
  const prevUnit = units[unitArrIndex - 1];

  // Update chunk index when query param changes
  useEffect(() => {
    const chunkParam = router.query.chunk;
    if (typeof chunkParam === 'string') {
      const chunkIndex = parseInt(chunkParam, 10);
      if (!Number.isNaN(chunkIndex) && chunkIndex >= 0 && chunkIndex < chunks.length) {
        setCurrentChunkIndex(chunkIndex);
      }
    }
  }, [router.query.chunk, chunks.length]);

  const handleChunkSelect = (index: number) => {
    setCurrentChunkIndex(index);
    router.push({
      pathname: router.pathname,
      query: { ...router.query, chunk: index },
    }, undefined, { shallow: true });
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  const handlePrevClick = () => {
    if ((isFirstChunk || chunks.length === 0) && prevUnit) {
      // Navigate to last chunk of previous unit
      router.push(`${prevUnit.path}?chunk=${(prevUnit.chunks?.length ?? 0) - 1}`);
    } else if (!isFirstChunk) {
      // Navigate to previous chunk
      handleChunkSelect(currentChunkIndex - 1);
    }
  };

  const handleNextClick = () => {
    if ((isLastChunk || chunks.length === 0) && nextUnit) {
      // Navigate to first chunk of next unit
      router.push(`${nextUnit.path}?chunk=0`);
    } else if (!isLastChunk) {
      // Navigate to next chunk
      handleChunkSelect(currentChunkIndex + 1);
    }
  };

  if (!unit || unitArrIndex === -1) {
    // Should never happen
    throw new Error('Unit not found');
  }

  return (
    <div>
      <Head>
        <title>{`${unit.courseTitle}: Unit ${unitNumber}`}</title>
        <meta name="description" content={unit.title} />
      </Head>

      <Breadcrumbs
        className="unit__breadcrumbs hidden md:block md:sticky md:top-16 z-10"
        route={{
          title: unit.courseTitle,
          url: unit.coursePath,
          parentPages: [ROUTES.home, ROUTES.courses],
        }}
      >
        <div className="unit__breadcrumbs-cta-container flex flex-row items-center gap-6">
          <button
            type="button"
            className="unit__breadcrumbs-cta flex flex-row items-center gap-1 no-underline cursor-pointer hover:text-color-primary disabled:opacity-50 disabled:hover:text-inherit disabled:cursor-not-allowed"
            disabled={isFirstChunk && !prevUnit}
            onClick={handlePrevClick}
            aria-label="Previous"
          >
            <FaArrowLeft className="size-3" /> Prev
          </button>
          <button
            type="button"
            className="unit__breadcrumbs-cta flex flex-row items-center gap-1 no-underline cursor-pointer hover:text-color-primary disabled:opacity-50 disabled:hover:text-inherit disabled:cursor-not-allowed"
            disabled={isLastChunk && !nextUnit}
            onClick={handleNextClick}
            aria-label="Next"
          >
            Next <FaArrowRight className="size-3" />
          </button>
        </div>
      </Breadcrumbs>

      <MobileHeader
        chunks={chunks}
        currentChunkIndex={currentChunkIndex}
        className="unit__mobile-header md:hidden sticky top-16 z-10"
        unit={unit}
        prevUnit={prevUnit}
        nextUnit={nextUnit}
        onPrevClick={handlePrevClick}
        onNextClick={handleNextClick}
      />

      <Section className="unit__main !border-none">
        <div className="unit__content-container flex flex-col md:flex-row">
          <SideBar
            courseTitle={unit.courseTitle}
            className="hidden md:block md:fixed md:overflow-y-scroll md:max-h-[calc(100vh-57px-65px-42px)]" // Adjust for Nav, Breadcrumb, and padding heights
            units={units}
            currentUnitNumber={unitNumber}
            chunks={chunks}
            currentChunkIndex={currentChunkIndex}
            onChunkSelect={handleChunkSelect}
          />
          <div className="unit__content flex flex-col flex-1 max-w-[680px] mx-auto gap-6 px-0 md:px-spacing-x md:ml-[320px]">
            <div className="unit__title-container">
              <P className="unit__course-title text-size-sm mb-2">Unit {unit.unitNumber}</P>
              <H1 className="unit__title font-serif text-[32px]">{chunks[currentChunkIndex]?.chunkTitle}</H1>
            </div>
            <MarkdownExtendedRenderer>
              {chunks[currentChunkIndex]?.chunkContent || unit.content}
            </MarkdownExtendedRenderer>

            {isLastChunk && (
              <UnitFeedback unit={unit} />
            )}

            {(!nextUnit && isLastChunk) ? (
              <>
                <Congratulations courseTitle={unit.courseTitle} coursePath={unit.coursePath} />
                <CertificateLinkCard courseId={unit.courseId} />
                <div className="unit__last-unit-cta-container flex flex-row justify-between mx-1">
                  <CTALinkOrButton className="last-unit__cta-link mx-auto" url={unit.coursePath} variant="secondary">
                    Back to course
                  </CTALinkOrButton>
                </div>
              </>
            ) : (
              <div className="unit__cta-container flex flex-row justify-between mt-6 mx-1">
                <CTALinkOrButton
                  className="unit__cta-link ml-auto"
                  onClick={handleNextClick}
                  variant="primary"
                  withChevron
                >
                  Complete and continue
                </CTALinkOrButton>
              </div>
            )}
          </div>
        </div>
      </Section>
    </div>
  );
};

export default UnitLayout;
