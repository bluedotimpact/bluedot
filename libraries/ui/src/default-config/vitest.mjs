/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
