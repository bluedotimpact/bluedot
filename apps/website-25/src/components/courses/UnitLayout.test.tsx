import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import UnitLayout from './UnitLayout';

const COURSE_UNITS = [
  {
    courseId: 'rec8CeVOWU0mGu2Jf',
    courseTitle: 'What the fish [Test Course]',
    coursePath: '/courses/test-course',
    path: '/courses/test-course/1',
    title: 'Basic Principles of Fish',
    content: "## Welcome to the deep end! \nThis unit explores the core concepts that _every_ fish enthusiast needs to understand, from basic anatomy to the physics of water movement.\nWe'll explore how fish \n[ ] communicate,\n[ ] navigate,\n[ ] and survive in their watery world\n...with special attention to the unique adaptations that make each species special.\nThrough hands-on exercises and detailed study materials, you'll develop a strong foundation in **_[ichthyology](https://en.wikipedia.org/wiki/Ichthyology)_** (that's fish science for those who don't speak fluent marine biologist).\n",
    duration: 12,
    unitNumber: '1',
    description: 'Master the fundamental concepts of fish biology, behavior, and basic aquatic principles.',
    id: 'recySscaN1b0Cm1jn',
  },
  {
    courseId: 'rec8CeVOWU0mGu2Jf',
    courseTitle: 'What the fish [Test Course]',
    coursePath: '/courses/test-course',
    path: '/courses/test-course/2',
    title: 'What Fish Are People Catching, and Why?',
    content: "Ever wondered why some fishermen chase tuna while others pursue cod? This unit dives into the complex world of modern fishing practices, from traditional line-fishing to industrial-scale operations. We'll examine the technological innovations driving the industry, the economic forces at play, and the eternal question of why that one weird fish at the bottom of the ocean looks like that. Through case studies and practical examples, you'll understand the delicate balance between feeding humanity and keeping our oceans stocked with sufficiently sassy fish.\n",
    duration: 5,
    unitNumber: '2',
    description: 'Explore current trends in fishing technology and the motives behind different fishing approaches.',
    id: 'recyGMcsDLhp9mPqH',
  },
  {
    courseId: 'rec8CeVOWU0mGu2Jf',
    courseTitle: 'What the fish [Test Course]',
    coursePath: '/courses/test-course',
    path: '/courses/test-course/3',
    title: 'The Promise of Fish',
    content: "Fish: they're not just for dinner anymore. This unit explores the untapped potential of our scaly friends in solving global challenges. From fish-powered renewable energy to underwater real estate development, we'll examine how piscine innovations are reshaping our world. We'll also delve into fish-based social networks (Fishbook), fish-inspired fashion trends, and why salmon make excellent financial advisors. Special attention will be paid to the emerging field of fish-human diplomacy.\n",
    duration: 110,
    unitNumber: '3',
    description: 'Understand how fish will revolutionize everything from the economy to your social life.',
    id: 'recjPDtSupcowWbmw',
  },
  {
    courseId: 'rec8CeVOWU0mGu2Jf',
    courseTitle: 'What the fish [Test Course]',
    coursePath: '/courses/test-course',
    path: '/courses/test-course/4',
    title: 'The Risks of Fish',
    content: "Not all that glitters is goldfish. This unit examines the darker side of piscine potential, from the threat of fish becoming too self-aware to the risks of sharks learning to use smartphones. Through careful analysis and possibly paranoid speculation, we'll prepare for worst-case scenarios like fish developing opposable fins or deciding to implement their own cryptocurrency. Includes emergency protocols for dealing with fish uprising scenarios.\n",
    duration: 8,
    unitNumber: '4',
    description: 'Explore what could go wrong when fish gain too much power.',
    id: 'recc6cSN9fn3VfTvZ',
  },
  {
    courseId: 'rec8CeVOWU0mGu2Jf',
    courseTitle: 'What the fish [Test Course]',
    coursePath: '/courses/test-course',
    path: '/courses/test-course/5',
    title: 'Contributing to Fish Safety',
    content: "### The future of fish-human relations depends on you! \nThis capstone unit brings together everything we've learned about our aquatic [overlords-in-waiting](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTBWqLpd7QgEtYIur4g48rq8PhAApbpOuO2fwoRB1aPvZpijMnS) and channels it into practical action. \nWe'll explore how to make the world a safer place for fish while maintaining appropriate species boundaries. \nThrough \n- case studies,\n- hands-on projects,\n- and possibly underwater meditation sessions,\n\nyou'll learn to bridge the gap between gill and lung breathers. \n\n> Warning: May cause increased empathy for tuna sandwiches.\n",
    duration: 15,
    unitNumber: '5',
    description: 'Learn how you can help create a better world for fish (and their reluctant human neighbors).',
    id: 'recIztjGqTNNpLTe1',
  },
];

describe('UnitLayout', () => {
  test('renders first unit as expected', () => {
    const { container } = render(
      <UnitLayout
        unit={COURSE_UNITS[0]!}
        unitNumber={1}
        units={COURSE_UNITS}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders previous and next unit buttons for middle unit', () => {
    const { container } = render(
      <UnitLayout
        unit={COURSE_UNITS[1]!}
        unitNumber={2}
        units={COURSE_UNITS}
      />,
    );
    expect(container.querySelector('.unit__cta-link')).toMatchSnapshot();
  });

  test('renders Congratulations section on final unit', () => {
    const { container } = render(
      <UnitLayout
        unit={COURSE_UNITS[COURSE_UNITS.length - 1]!}
        unitNumber={COURSE_UNITS.length}
        units={COURSE_UNITS}
      />,
    );
    expect(container.querySelector('.unit__cta-container')).toBeNull();
    expect(container.querySelector('.last-unit-cta-container')).toBeTruthy();
    expect(container.querySelector('.congratulations')).toBeTruthy();
  });
});
