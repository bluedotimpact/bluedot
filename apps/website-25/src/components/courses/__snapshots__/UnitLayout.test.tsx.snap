// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnitLayout > renders Congratulations section on final unit 1`] = `
<div
  class="last-unit"
>
  <div
    class="congratulations flex flex-col gap-4 container-lined p-4 bg-white items-center"
  >
    <h3
      class="bluedot-h3 congratulations__title text-center"
    >
      Congratulations on completing 
      What the fish [Test Course]
      !
    </h3>
    <p
      class="bluedot-p congratulations__description text-center"
    >
      Now share your perspective! Those reflections aren't going to achieve much sitting in an exercise box. Share them with your network to get the conversation going.
    </p>
    <div
      class="social-share flex flex-row gap-4"
    >
      <a
        class="social-share__link size-6"
        href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dlinkedin&text=🎉 I just completed the What the fish [Test Course] course from BlueDot Impact! It’s free, self-paced, and packed with insights. Check it out and sign up with my referral link below:"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="social-share__link-icon size-6"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
          />
        </svg>
      </a>
      <a
        class="social-share__link size-6"
        href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dtwitter&text=🎉 I just completed the What the fish [Test Course] course from BlueDot Impact! It’s free, self-paced, and packed with insights. Check it out and sign up with my referral link below:"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="social-share__link-icon size-6"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
          />
        </svg>
      </a>
      <a
        class="social-share__link size-6"
        href="https://www.facebook.com/sharer/sharer.php"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="social-share__link-icon size-6"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 256C512 114.6 397.4 0 256 0S0 114.6 0 256C0 376 82.7 476.8 194.2 504.5V334.2H141.4V256h52.8V222.3c0-87.1 39.4-127.5 125-127.5c16.2 0 44.2 3.2 55.7 6.4V172c-6-.6-16.5-1-29.6-1c-42 0-58.2 15.9-58.2 57.2V256h83.6l-14.4 78.2H287V510.1C413.8 494.8 512 386.9 512 256h0z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    class="last-unit__cta-container flex flex-row justify-between mt-6 mx-1"
  >
    <a
      class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter last-unit__cta-link mx-auto"
      data-testid="cta-link"
      href="/courses/test-course"
    >
      <span
        class="cta-button__text"
      >
        Back to course
      </span>
    </a>
  </div>
</div>
`;

exports[`UnitLayout > renders first unit as expected 1`] = `
<div>
  <div>
    <div
      class="hero-section bg-bluedot-darker flex flex-row justify-center items-center w-full px-spacing-x unit__hero"
    >
      <div
        class="hero-section__content max-w-[865px] py-12"
      >
        <div
          class="hero-section__mini-title text-on-dark text-center uppercase tracking-wider text-sm font-semibold mb-4"
        >
          What the fish [Test Course]
        </div>
        <h1
          class="hero-section__title text-on-dark text-center bluedot-h1"
        >
          Basic Principles of Fish
        </h1>
      </div>
    </div>
    <div
      class="breadcrumbs bg-color-canvas border-b border-color-divider w-full py-space-between unit__breadcrumbs sticky top-[72px] md:top-[100px] z-10"
    >
      <nav
        aria-label="Breadcrumbs"
        class="breadcrumbs__nav section-base"
      >
        <ol
          class="breadcrumbs__list flex"
        >
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="breadcrumbs__link bluedot-a no-underline"
              href="/"
            >
              Home
            </a>
            <span
              class="breadcrumbs__separator mx-2"
            >
              &gt;
            </span>
          </li>
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="breadcrumbs__link bluedot-a no-underline"
              href="/courses"
            >
              Courses
            </a>
            <span
              class="breadcrumbs__separator mx-2"
            >
              &gt;
            </span>
          </li>
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="breadcrumbs__link bluedot-a no-underline"
              href="/courses/test-course"
            >
              What the fish [Test Course]
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <section
      class="section section-body unit__main"
    >
      <div
        class="section__body"
      >
        <div
          class="unit__content-container flex flex-col md:flex-row gap-16"
        >
          <div
            class="sidebar w-full md:w-[332px] flex flex-col"
          >
            <div
              class="sidebar__content flex flex-col gap-6"
            >
              <div
                class="sidebar__unit border sidebar__unit--active border-color-primary rounded-lg p-4 flex flex-col gap-2"
              >
                <div
                  class="sidebar__unit-item flex flex-col gap-2"
                >
                  <p
                    class="bluedot-p sidebar__title uppercase text-size-xs font-bold"
                  >
                    Unit 
                    1
                  </p>
                  <a
                    class="bluedot-a sidebar__description bluedot-h4 no-underline"
                    href="/courses/test-course/1"
                  >
                    Basic Principles of Fish
                  </a>
                  <p
                    class="bluedot-p sidebar__description text-size-s"
                  >
                    Master the fundamental concepts of fish biology, behavior, and basic aquatic principles.
                  </p>
                </div>
              </div>
              <div
                class="sidebar__unit border border-color-divider rounded-lg p-4 flex flex-col gap-2"
              >
                <div
                  class="sidebar__unit-item flex flex-col gap-2"
                >
                  <p
                    class="bluedot-p sidebar__title uppercase text-size-xs font-bold"
                  >
                    Unit 
                    2
                  </p>
                  <a
                    class="bluedot-a sidebar__description bluedot-h4 no-underline"
                    href="/courses/test-course/2"
                  >
                    What Fish Are People Catching, and Why?
                  </a>
                  <p
                    class="bluedot-p sidebar__description text-size-s"
                  >
                    Explore current trends in fishing technology and the motives behind different fishing approaches.
                  </p>
                </div>
              </div>
              <div
                class="sidebar__unit border border-color-divider rounded-lg p-4 flex flex-col gap-2"
              >
                <div
                  class="sidebar__unit-item flex flex-col gap-2"
                >
                  <p
                    class="bluedot-p sidebar__title uppercase text-size-xs font-bold"
                  >
                    Unit 
                    3
                  </p>
                  <a
                    class="bluedot-a sidebar__description bluedot-h4 no-underline"
                    href="/courses/test-course/3"
                  >
                    The Promise of Fish
                  </a>
                  <p
                    class="bluedot-p sidebar__description text-size-s"
                  >
                    Understand how fish will revolutionize everything from the economy to your social life.
                  </p>
                </div>
              </div>
              <div
                class="sidebar__unit border border-color-divider rounded-lg p-4 flex flex-col gap-2"
              >
                <div
                  class="sidebar__unit-item flex flex-col gap-2"
                >
                  <p
                    class="bluedot-p sidebar__title uppercase text-size-xs font-bold"
                  >
                    Unit 
                    4
                  </p>
                  <a
                    class="bluedot-a sidebar__description bluedot-h4 no-underline"
                    href="/courses/test-course/4"
                  >
                    The Risks of Fish
                  </a>
                  <p
                    class="bluedot-p sidebar__description text-size-s"
                  >
                    Explore what could go wrong when fish gain too much power.
                  </p>
                </div>
              </div>
              <div
                class="sidebar__unit border border-color-divider rounded-lg p-4 flex flex-col gap-2"
              >
                <div
                  class="sidebar__unit-item flex flex-col gap-2"
                >
                  <p
                    class="bluedot-p sidebar__title uppercase text-size-xs font-bold"
                  >
                    Unit 
                    5
                  </p>
                  <a
                    class="bluedot-a sidebar__description bluedot-h4 no-underline"
                    href="/courses/test-course/5"
                  >
                    Contributing to Fish Safety
                  </a>
                  <p
                    class="bluedot-p sidebar__description text-size-s"
                  >
                    Learn how you can help create a better world for fish (and their reluctant human neighbors).
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="unit__content flex flex-col flex-1 max-w-[728px] gap-4"
          >
            <div
              class="markdown-extended-renderer prose"
            />
            <div
              class="unit__cta-container flex flex-row justify-between mt-6 mx-1"
            >
              <a
                class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--primary bg-bluedot-normal link-on-dark unit__cta-link ml-auto"
                data-testid="cta-link"
                href="/courses/test-course/2"
              >
                <span
                  class="cta-button__text"
                >
                  Next unit
                </span>
                <span
                  class="cta-button__chevron ml-3"
                >
                  <img
                    alt="→"
                    class="cta-button__chevron-icon size-2"
                    src="/icons/chevron_white.svg"
                  />
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`UnitLayout > renders previous and next unit buttons for middle unit 1`] = `
<a
  class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter unit__cta-link mr-auto"
  data-testid="cta-link"
  href="/courses/test-course/1"
>
  <span
    class="cta-button__chevron mr-3"
  >
    <img
      alt="←"
      class="cta-button__chevron-icon size-2 rotate-180"
      src="/icons/chevron_blue.svg"
    />
  </span>
  <span
    class="cta-button__text"
  >
    Previous unit
  </span>
</a>
`;
